lib/chef/provider/package/snap.rb

```diff
@@ -311,6 +349,9 @@ def get_latest_package_version(name, channel)
             raise Chef::Exceptions::Package, json["result"], caller
           end
 
+          unless json["result"][0]["channels"]["latest/#{channel}"]
+            fail "No version of #{name} in channel #{channel}"
```

Comment:
Since `fail` is a synonym of `raise`, should this use a more specific exception class like the `raise` a few lines up?

Reply:
yes it should. Fixed.